pyspecProcScripts/table_integrals.py

```diff
@@ -0,0 +1,254 @@
+        Real=False,
+        alias_slop=3,
+        clock_correction = True,
+        error_bars = True,
+        correlate = True,
+        fl=None):
+    """Applies appropriate phase corrections, along with DC offset corrections, 
+    alignment, and integrates the data to transform 2D datasets into a table
+    of integrals.
+
+    Parameters
+    ==========
+    searchstr:      str
+                    string for the title of the plots produced.
+    signal_pathway: dict
+                    Dictionary of the path of the desired signal.
+    excluded_pathways:  dict
+                        Dictionary of all coherence pathways that are not 
+                        the signal pathway.
+    f_range:        tuple
+                    Frequency range over which the signal resides.
+    t_range:        tuple
+                    Time range over which the signal resides.
+    sgn:            nddata
+                    A dataset with all +1 or -1 (giving the sign of the original signal)
+                    Does *not* include the 'direct' dimension.
+    direct:         str
+                    Direct dimension of the signal
+    indirect:       str
+                    Indirect dimension of the signal. Usually 'power' or 'nScans' 
+                    for example.
+    Real:           bool
+                    If true applies power axis corrections as it is real data.
+    alias_slop:     int
+                    Aliasing_slop used in the hermitian function.
+    clock_correction:   bool
+                        If true, will apply a clock correction. Is needed for IR
+                        but not enhancement data.
+    error_bars:     bool
+                    Option to apply integration *with* error bars.
+    correlate:      bool
+                    Option to apply correlation alignment to data.
+            
+    Returns
+    =======
+    s:              nddata
+                    Data with applied corrections but *not* integrated
+    s_int:          nddata
+                    Integrated and corrected data
+    """                
+    signal_keys = list(signal_pathway)
+    signal_values = list(signal_pathway.values())
+    s.ift(direct)
+    if Real:
+        p_axis = s.getaxis('power')
+        power_axis_dBm = array(s.get_prop('meter_powers'))
+        power_axis_W = zeros_like(power_axis_dBm)
+        power_axis_W[:] = (1e-2*10**((power_axis_dBm[:]+10.)*1e-1))
+        power_axis_W = r_[0,power_axis_W]
+    s.reorder([indirect,direct],first=False)
+    #{{{DC offset correction
+    s.ift(list(signal_pathway))
+    t_start = t_range[-1] / 4
+    t_start *= 3
+    rx_offset_corr = s['t2':(t_start,None)]
+    rx_offset_corr = rx_offset_corr.data.mean()
+    s -= rx_offset_corr
+    s.ft(direct)
+    s.ft(list(signal_pathway))
+    #}}}
+    zero_crossing = abs(select_pathway(s[direct:f_range],signal_pathway)).C.sum(direct).argmin(indirect,raw_index=True).item()
+    print("ZERO CROSSING IS",zero_crossing)
```

Comment:
we have the function that determiens the sign of the signal --  why is it doing this?